### ISSUE TITLE
FR:  grey out the "reference pages" menu option if there are no reference pages, rather than omit the menu item

### ISSUE THREAD

**Issue by p-phillips** [koreader/koreader#12893]
I've noticed a few posts recently where folk have struggled to remove the reference pages from the RH margin.  It can be tricky and confusing explaining where this menu item is and how to turn it off if a book doesn't have reference pages.

**My suggestion is to grey out the reference pages menu item rather than just not show it.**  

I know that the likelihood that people looking for info on how to turn the ref pages off will presumably be reading a book **with** reference pages, but if they are like me, flipping between books, that may not be the case.   

I would not think that whether a menu item was present depended on the book. So I could go looking in the menus and not be able to find the menu item.

**Comment by poire-z** [comment]
The reasoning about not showing it was that books with reference pages are rare, and it gives room for another more important item at the bottom in this menu (that would be put on the 2nd page if Reference pages was greyed out).
And we/I still want this Reference page in the first menu, so that it jumps at you when you meet a book with Reference pages.

**Comment by p-phillips** [comment]
I don't think they are that rare - I've only been using my Kobo for about 6 weeks and have come across 3-4 books with ref pages. And the first time I saw them I thought the formatting of my book was screwed up.

I understand your reasoning, but it is very confusing to have menu items which someone can't demonstrate without the right book.  What about compressing the menu layout a bit so it doesn't bump the other items onto a second page?

**Comment by Frenzie** [comment]
If it were quite so easy it'd have been done already. :-)

Fwiw, I'm not sure if I've ever seen any in the wild, besides explicit sample/testing books.

**Comment by jonnyl2** [comment]
Reference pages are quite common in Gutenberg.org books (which needless to say are very popular). 

I don't really have an opinion on whether the Reference pages menu should always be visible, but as far as condensing the menu goes -- do we really need the `Go back to previous location` and `Go forward to next location` in the top menu? These are quick navigation options perfect for gestures that I don't think anyone would want to access via the top menu, but they take 2 spots at this prime menu location. (Or as a compromise, just make it "Go back.." and assign "Go forward.." to long-press.)

**Comment by Frenzie** [comment]
> Reference pages are quite common in Gutenberg.org books (which needless to say are very popular).

That must be a recent addition. Nice!

**Comment by poire-z** [comment]
> I don't think they are that rare - I've only been using my Kobo for about 6 weeks and have come across 3-4 books with ref pages.

Depends on what you read, the origin of you books, your country publisher habits... Some people may always see it, some never, me about 30%.

> And the first time I saw them I thought the formatting of my book was screwed up.

So, you saw them, and you had this menu, and were able to understand what it was about. May be it should be mentionned in our user guide (if it is not yet).
You didn't need to see it when you were not thinking the formatting of your book was screwed up :)

> I understand your reasoning, but it is very confusing to have menu items which someone can't demonstrate without the right book.

Well, we already have different menu items when reading an EPUB vs. when reading a PDF: we get the menu items that work and make sense in the context. We don't keep all the "other context" related menu items grey. People reading EPUB don't need to be bothered with grey PDF-related items.
It's a little less strong with EPUB with reference pages, but the same idea works.

Moreover, the way we use disabled/grey menu items is usually that some other combination of other settings will make them enabled/not-gray, If you see something gray, you can assume you can get it enabled by playing around.
For reference pages, no matter what you do, if the EPUB does not have them, you will never get that menu item enabled.

> What about compressing the menu layout a bit so it doesn't bump the other items onto a second page?

That first menu tab is one of the most dynamic: some plugins may put stuff there, Non-Touch will get more, so in some context, it will be 2 pages and it's not worth fighting the issue. We can only try to make the most valuable first ("go back to previous location" is more useful than "go forward").

> do we really need the `Go back to previous location` and `Go forward to next location` in the top menu? 
These are quick navigation options perfect for gestures that I don't think anyone would want to access via the top menu, but they take 2 spots at this prime menu location. (Or as a compromise, just make it "Go back.." and assign "Go forward.." to long-press.)

I think we need them - with no long-press trick (we used the long-press trick for "Book map (overview)" to avoid taking a slot, but it's a little bad :) not too bad because it's not a super different feature than the regular Book map and a user can be fine with only the regular one).
Some people may not be bothered with using gestures (don't like them, can't use them): they should have access to everything via menu.
And people using gestures should be fine with not using the menu, or rarely need to get to it, so they should be the less bothered with it being big and multi pages :)

As for me, I have "go back to previous location" via a gesture. I don't have "go forward" set to a gesture, because I normally never use it, and don't want to have to waste a gesture (that i will not remember for lack of using it). BUT I may accidentally "go back" with my gesture, and a few times I wanted to go forward, and I was happy knowing and having it available in the menu.

**Comment by offset-torque** [comment]
I just wrote a **"Reference Pages"** section for the guide (before seeing this issue). It will be added soon. At least this will help explaining the feature to the new users and can be referenced for future questions.

**Comment by p-phillips** [comment]
> > And the first time I saw them I thought the formatting of my book was screwed up.
 
> So, you saw them, and you had this menu, and were able to understand what it was about. 
Yeah, but i'm a tech person.  I'm probably going to go delving around (and looking on github) unlike your average user.

> **May be it should be mentioned in our user guide (if it is not yet)**. 
It's not in the user guide - I had already checked.  I think that would help.

I think the other issue is that most new users wouldn't know that those margin numbers are actually known as "reference pages", so perhaps something could be added  to the images on the Reading Screen section of the user guide with an example of how reference pages display ?    Also, perhaps n addition to the trouble shooting section at the end "Why am I seeing digits on the RHS margin of my pages ?" ???

Anyway, thanks for engaging on the issue and of course, thanks for a neat program.

**Comment by offset-torque** [comment]
> It's not in the user guide - I had already checked.

You missed the "It will be added soon" part of my reply **p-phillips** :)
Our guide is updated in big batches so it takes time.

> I think the other issue is that most new users wouldn't know that those margin numbers are actually known as "reference pages", so perhaps something could be added to the images on the Reading Screen section of the user guide with an example of how reference pages display ? Also, perhaps n addition to the trouble shooting section at the end "Why am I seeing digits on the RHS margin of my pages ?" ???

Exactly my thoughts. I was thinking of adding a screenshot of the "Reading Screen" which shows all our different indicators like highlight, partial refresh etc. and also reference numbers. Troubleshooting section is a good idea too. I will add it there also.

**Comment by p-phillips** [comment]
> > It's not in the user guide - I had already checked.
> 
> You missed the "It will be added soon" part of my reply **p-phillips** :) 

I'm going to shout "foul" here!  That must have been posted when I was composing my own post!   :-)  

Anyway that sounds good to me.    So I think the issue can be closed unless @offset-torque wants to keep it open until the guide is updated ?

**Comment by offset-torque** [comment]
I made notes to update the (already written but not published yet) section according to our conversation. I track user guide tasks independently, so I don't need this issue to stay open.
